internal/store/node.go

```diff
@@ -40,341 +40,377 @@ var (
 
 func nodeMetricFamilies(allowLabelsList []string) []generator.FamilyGenerator {
 	return []generator.FamilyGenerator{
-		*generator.NewFamilyGenerator(
-			"kube_node_info",
-			"Information about a cluster node.",
-			metric.Gauge,
-			"",
-			wrapNodeFunc(func(n *v1.Node) *metric.Family {
-				labelKeys := []string{
-					"kernel_version",
-					"os_image",
-					"container_runtime_version",
-					"kubelet_version",
-					"kubeproxy_version",
-					"provider_id",
-					"pod_cidr",
-				}
-				labelValues := []string{
-					n.Status.NodeInfo.KernelVersion,
-					n.Status.NodeInfo.OSImage,
-					n.Status.NodeInfo.ContainerRuntimeVersion,
-					n.Status.NodeInfo.KubeletVersion,
-					n.Status.NodeInfo.KubeProxyVersion,
-					n.Spec.ProviderID,
-					n.Spec.PodCIDR,
-				}
+		nodeCreatedGenerator(),
```

Comment:
I assume you called the function Generator suffix due to it returning a FamilyGenerator type?

Still think might make sense to call them `createNodeCreateFamilyGenerator`, but not a blocker for me, wdyt?

Reply:
Agreed, I thought a getter like naming approach would be better here, but after giving it another thought, these functions are actual constructors so I prefer your suggestion.

Reply:
Nice, thanks!